### PR TITLE
Ghosttest

### PR DESCRIPTION
**Pull request by nicolasaunai**
#246 on top of #250

### CODE REVIEW

File: Testing/Temporary/CTestCostData.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+---
```
Comment (Dekken) [inline Testing/Temporary/CTestCostData.txt:1]:
```
git rm -r --cached Testing
```

File: pyphare/pyphare/pharesee/hierarchy.py [hunk 2]
```diff
@@ -337,10 +342,10 @@ def hierarchy_fromh5(h5_filename, time, hier):
         # then add all other times
         print("creating hierarchy from all times in file")
         times = list(data_file.keys())
-        hier = hierarchy_from(h5_filename, time=times[0])
+        hier = hierarchy_from(h5_filename=h5_filename, time=times[0])
```
[the remark this answers is not in the document]
Comment (nicolasaunai) [reply pyphare/pyphare/pharesee/hierarchy.py:345]: it does not.
```
hierarchy_fromh5(h5_filename, time, hier)
```

File: pyphare/pyphare/pharesee/hierarchy.py [hunk 3]
```diff
@@ -537,11 +541,15 @@ def hierarchy_from_sim(simulator, sim_hier, qty, pop=""):
                 lower = int(patch.lower[0])
                 upper = int(patch.upper[0])
                 origin = float(patch.origin)
-                layout = GridLayout(Box(lower, upper), origin, lvl_cell_width)
+                layout = GridLayout(Box(lower, upper), origin, lvl_cell_width, interp_order=simulator.interp_order())
+
+                v = np.asarray(patch.data.v)
```
[the remark this answers is not in the document]
Comment (nicolasaunai) [reply pyphare/pyphare/pharesee/hierarchy.py:546]: that's enough indeed

File: Testing/Temporary/CTestCostData.txt (hunk 1 above)
Comment (nicolasaunai) [reply Testing/Temporary/CTestCostData.txt:1]: done

File: tests/simulator/test_init_ghostparticles.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+                # the overlap box overlaps both ghost and domain cells
+                # we need to extract the domain ones to later select domain
+                # particles
+                ovlped_refdom = ref_pd.box * shift_refbox
+                ovlped_cmpdom = cmp_pd.box * shift_cmpbox
+
+                # on lvl 0 patches are adjacent
+                # therefore the overlap box must overlap the
+                # patchData box. 1 cell in interporder1, 2 cells for higher
+                #self.assertIsNotNone(ovlped_refdom)
```
Comment (Dekken) [inline tests/simulator/test_init_ghostparticles.py:138]: needed?

File: tests/simulator/test_init_ghostparticles.py [hunk 5]
```diff
@@ -0,0 +1,177 @@
+
+from pybindlibs import cpp
```
Comment (Dekken) [inline tests/simulator/test_init_ghostparticles.py:2]: possibly unused

File: Testing/Temporary/CTestCostData.txt (hunk 1 above)
Comment (Dekken) [reply Testing/Temporary/CTestCostData.txt:1]: files are still here

File: tests/simulator/test_init_ghostparticles.py [hunk 6]
```diff
@@ -109,7 +126,7 @@ def test_patch_ghost_particle_are_clone_of_overlaped_patch_domain_particles(self
                 if ilvl != 0: #only root level tested here
                     continue
 
-                if overlap["pdatas"][0].quantity != "particles":
+                if "particles"  not in overlap["pdatas"][0].quantity :
```
Comment (Dekken) [inline tests/simulator/test_init_ghostparticles.py:129]: double space :eyes:

**Comment by Dekken** [comment]
I had commits in this PR once

File: src/python3/cpp_simulator.cpp [hunk 7]
```diff
@@ -51,6 +51,9 @@ void setPatchData(PatchData& data, std::string patchID, std::string origin,
     std::memcpy(data.upper.request().ptr, upper.data(), bytes);
     data.patchID = patchID;
     data.origin  = origin;
+
+    std::memcpy(data.lower.request().ptr, lower.data(), PatchData::dimension);
```
Comment (Dekken) [inline src/python3/cpp_simulator.cpp:55]: what's on master is correct 
```
https://github.com/PHAREHUB/PHARE/blob/master/src/python3/cpp_simulator.cpp#L49
```

Comment (Dekken) [reply src/python3/cpp_simulator.cpp:55]: conflict lines possibly?

File: tests/simulator/test_initialization.py [hunk 8]
```diff
@@ -0,0 +1,587 @@
...
+
+                xbx   = patch.patch_datas["EM_B_x"].x[:]
+                bx  = patch.patch_datas["EM_B_x"].dataset[:]
+                np.testing.assert_allclose(bx, bx_fn(xbx), atol=3e-5)
+
+                xby   = patch.patch_datas["EM_B_y"].x[:]
+                by  = patch.patch_datas["EM_B_y"].dataset[:]
+                np.testing.assert_allclose(by, by_fn(xby), atol=3e-5)
+
+                xbz   = patch.patch_datas["EM_B_z"].x[:]
```
Comment (Dekken) [inline tests/simulator/test_initialization.py:196]: not directly related to this PR (I think), but should this
```
https://github.com/nicolasaunai/PHARE/blob/ghosttest/pyphare/pyphare/pharesee/hierarchy.py#L58
```

be a property?

Comment (nicolasaunai) [reply tests/simulator/test_initialization.py:196]: maybe, but we do not necessarily want to compute that each time

File: src/python3/cpp_simulator.cpp (hunk 7 above)
Comment (nicolasaunai) [reply src/python3/cpp_simulator.cpp:55]: yeah conflict in rebase poorly handled by me :( fixed, thx

**Comment by Dekken** [comment]
should be good to merge once rebased to master for the CMake MPI build fix

**Comment by nicolasaunai** [comment]
replaced by #286

**Pull request closed without merging**